### DISCUSSION TITLE
December release - New REST API connector + home screen redesign

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#4120] [Announcements]
Previous to this release, the REST API data source was shoe-horned into a generic data source template. This was not ideal, so in December and early January, the team worked tremendously hard to improve and deliver a best-in-class REST API experience. We collectively feel we have delivered. But, ultimately, it's your opinion that matters, so please, let us know what you think. 

A lot of you will have noticed a new home screen design. This is still a work in progress, but in our opinion, it's already a huge improvement from a visual and usability perspective. 

Below, you can find more information regarding the release:

**REST API connector**
Use the new powerful Rest API connector to connect with REST APIs, fetch data, receive form data, authenticate with external APIs, and build modern business apps in minutes.

![rest-api-2](https://user-images.githubusercontent.com/49767913/150331526-294fe91a-5b00-4831-bcb1-b193eb15b38f.png)

**Curl and Swagger REST query importer**
The REST importer allows you to use an existing Swagger collection or cURL request to create new queries. Also, you can import queries into an existing data source, or use the import to create a new data source.

**New home screen design**
We redesigned the Budibase home screen to accommodate more templates and provide users with the ability to assign icons to apps to improve usability. You can add your feedback here, or #4112 

![homescreen-ui-2](https://user-images.githubusercontent.com/49767913/150331549-786fa72f-91c7-47c7-ab1a-87441823574a.png)

**Duplicate queries**
You can now save time by duplicating queries within your data section.

**Additional, important improvements**
We’ve removed the automation outgoing webhook in favor of using the REST connector directly.
You can now skip fetching tables for plus data sources

**Coming up in the January Release (~ 2 weeks time)**

- Google sheets integration #4089 
- JSON data field 
- New rich text component

**Comment by SantoshSrinivas79** [comment]
The changes in the REST API connector look great and the design changes are fantastic!

i) Could you please explain how the Dynamic Variables are to be used with an example?

ii) Anyway to handle any one of the below issues?

- https://github.com/Budibase/budibase/issues/3827
- https://github.com/Budibase/budibase/discussions/3934
- https://github.com/Budibase/budibase/discussions/3879

**Reply by mjashanks** [reply]
None of these are covered in the new REST Api. I agree that we should add them. I have collated these into one issue - there may be more things that we would like to add to this: https://github.com/Budibase/budibase/issues/4139

**Reply by Rory-Powell** [reply]
Hey @SantoshSrinivas79 , docs on dynamic variables are available here
https://docs.budibase.com/building-apps/data/tables/rest/rest-variables#dynamic-variables
